src/Blog_App_backend/main.mo

```diff
@@ -1,5 +1,77 @@
-actor {
-  public query func greet(name : Text) : async Text {
-    return "Hello, " # name # "!";
-  };
-};
+import Text "mo:base/Text";
+import Principal "mo:base/Principal";
+import Result "mo:base/Result";
+import HashMap "mo:base/HashMap";
+import Nat "mo:base/Nat";
+import Iter "mo:base/Iter";
+
+actor BlogApp {
+
+    type Post = {
+        title: Text;
+        description: Text;
+        upvotes: Nat;
+        downvotes: Nat;
+        isPublic: Bool;
+    };
+
+    stable var stablePosts : [(Principal, Post)] = [];
+
+    let posts = HashMap.HashMap<Principal, Post>(0, Principal.equal, Principal.hash);
+
+    ///Function for preUpgrade
+    system func preupgrade() {
+      stablePosts := Iter.toArray(posts.entries());
+    };
+
+    ///Function for postUpgrade
+    system func postupgrade() {
+      for ((p,post) in stablePosts.vals()){
+        posts.put(p,post);
+      };
+    };
+
+    ///Function to create posts
+    public shared({ caller }) func createPost(title: Text, description: Text): async Result.Result<(), Text> {
+        if (title == "" or description == "") {
+            return #err("Title and description must not be empty");
+        };
+
+        if (Text.size(title) > 125) {
+            return #err("Title must not exceed 125 characters");
+        };
+
+        if (Principal.isAnonymous(caller)){
+            return # err("Anonymous principle cannot create post");
+        };
```

Comment:
create a new function to validate the post and use it here

Reply:
ya done